kademlia/src/main/java/ingsw/group1/kademlia/NodeUtils.java

```diff
@@ -25,7 +25,7 @@
      * @return the binary key generated with SHA-1 hashing for the given peer
      * @throws IllegalArgumentException if key length or given peer are invalid or null
      */
-    public static BitSet getIdForPeer(Peer<String> peer, int keyLength) {
+    public static <P extends Peer<String, P>> BitSet getIdForPeer(P peer, int keyLength) {
```

Comment:
` <P extends Peer<String, P>>`  what?

Reply:
Since 5 days have passed I totally forgot what these fixes are for, I believe they are regarding the conversion of Peer to Interface, I added a generic P which extends Peer to the Peer interface itself, this way a class can decide to restrict which kinds of Peer it's comparable with. For example we now have:
`SMSPeer extends Peer<String, SMSPeer>`
The class can be declared in other ways for example:
`SMSPeer extends Peer<String, Peer<String,?>` to allow comparing with any Peer having a String key, but it would make little sense.
("?" Is the wildcard operator)

Reply:
In this method I'm saying the generic P extends Peer with a String address and which is comparable only to a Peer of the same type, it would be better to change to:
`<P extends Peer<String,?>>` Since we only care what the address is.

Reply:
OK, now I understand